Mage.Sets/src/mage/cards/s/Snowfall.java

```diff
@@ -0,0 +1,174 @@
+
+        // Cumulative upkeep {U}
+        this.addAbility(new CumulativeUpkeepAbility(new ManaCostsImpl<>("{U}")));
+
+        // Whenever an Island is tapped for mana, its controller may add an additional {U}.
+        // If that Island is snow, its controller may add an additional {U}{U} instead.
+        // Spend this mana only to pay cumulative upkeep costs.
+        this.addAbility(new TapForManaAllTriggeredManaAbility(new SnowfallManaEffect(), filter, SetTargetPointer.PERMANENT));
+    }
+
+    private Snowfall(final Snowfall card) {
+        super(card);
+    }
+
+    @Override
+    public Snowfall copy() {
+        return new Snowfall(this);
+    }
+}
+
+class SnowfallManaEffect extends ManaEffect {
+
+    private static final Set<String> choice = new HashSet<>();
+
+    static {
+        choice.add("Add one additional blue mana.");
+        choice.add("Add two additional blue mana.");
+        choice.add("Don't add additional mana.");
+    }
+
+    SnowfallManaEffect() {
+        super();
+        this.staticText = "its controller may add an additional {U}. " +
+                "If that Island is snow, its controller may add an additional {U}{U} instead. " +
+                "Spend this mana only to pay cumulative upkeep costs.";
+    }
+
+    private SnowfallManaEffect(final SnowfallManaEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public Player getPlayer(Game game, Ability source) {
+        Permanent land = getTargetPointer().getFirstTargetPermanentOrLKI(game, source);
+        if (land != null) {
+            return game.getPlayer(land.getControllerId());
+        }
+        return null;
+    }
+
+    @Override
+    public List<Mana> getNetMana(Game game, Ability source) {
+        List<Mana> netMana = new ArrayList<>();
+        if (game == null) {
+            return netMana;
+        }
+
+        Permanent land = getTargetPointer().getFirstTargetPermanentOrLKI(game, source);
+        if (land == null) {
+            return netMana;
+        }
+
+        Mana mana = Mana.BlueMana(land.isSnow(game) ? 2 : 1);
+        netMana.add(mana);
+        return netMana;
+    }
+
+    @Override
+    public Mana produceMana(Game game, Ability source) {
+        if (game == null) {
+            return new Mana();
+        }
+
+        Permanent land = getTargetPointer().getFirstTargetPermanentOrLKI(game, source);
+        if (land == null) {
+            return new Mana();
+        }
+        Player player = game.getPlayer(land.getControllerId());
+        if (player == null) {
+            return new Mana();
+        }
+
+        if (land.isSnow(game)) {
+            Choice choiceImpl = new ChoiceImpl(true);
+            choiceImpl.setMessage("Add additional blue mana? This mana can only be spent to pay cumulative upkeep costs.");
+            choiceImpl.setChoices(choice);
+
+            if (player.choose(Outcome.Benefit, choiceImpl, game)) {
```

Comment:
I agree that inheriting `Outcome.PutManaInPool` makes more sense. I originally did that, but it was causing a problem where, when tapping a snow Island to pay colorless mana, it would not ask about adding additional mana, nor would it actually add any of the additional mana that could be added.

Looking into this further, [ManaUtil.tryToAutoSelectAManaColor()](https://github.com/magefree/mage/blob/0470ae9799641ff556de848989049aa51089b9aa/Mage/src/main/java/mage/util/ManaUtil.java#L132C1-L132C1) is auto-selecting the first choice for the player.

I've put in a change to ensure that the game will only try to auto-select color choices for the player, like it's implemented for the computer [here](https://github.com/magefree/mage/blob/0470ae9799641ff556de848989049aa51089b9aa/Mage.Server.Plugins/Mage.Player.AI/src/main/java/mage/player/ai/ComputerPlayer.java#L1959C8-L1959C8). Please let me know if this is satisfactory.

Reply:
Good find.